tests/data/Compressor.cpp

```diff
@@ -0,0 +1,251 @@
+
+/* Copyright (c) 2010-2016, Cedric Stalder <cedric.stalder@gmail.com>
+ *                          Stefan Eilemann <eile@eyescale.ch>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 2.1 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#define TEST_RUNTIME 600 // seconds
+#include <lunchbox/test.h>
+
+#include <pression/data/Compressor.h>
+#include <pression/data/Registry.h>
+
+#include <lunchbox/buffer.h>
+#include <lunchbox/clock.h>
+#include <lunchbox/file.h>
+#include <lunchbox/memoryMap.h>
+#include <lunchbox/rng.h>
+
+#include <boost/program_options.hpp>
+#include <algorithm>
+
+using lunchbox::Strings;
+namespace po = boost::program_options;
+
+void _testFile( int argc, char** argv );
+void _testRandom();
+void _testData( const std::string& name, uint8_t* data, uint64_t size );
+void getFiles( Strings& files, const std::string& ext );
+
+pression::data::Registry& registry = pression::data::Registry::getInstance();
+uint64_t _result = 0;
+uint64_t _size = 0;
+float _compressionTime = 0;
+float _decompressionTime = 0;
+float _baseTime = 0.f;
+
+int main( const int argc, char** argv )
+{
+    _testFile( argc, argv );
+    _testRandom();
+    return EXIT_SUCCESS;
+}
+
+pression::data::CompressorInfos getCompressors()
+{
+    auto infos = registry.getInfos();
+    for( auto i = infos.begin(); i != infos.end(); ++i )
+        if( i->name == "pression::data::CompressorRLE" ) // move RLE to front
+        {
+            std::swap( *i, *infos.begin( ));
+            return infos;
+        }
+
+    return infos;
+}
+
+void _testData( const pression::data::CompressorInfo& info,
+                const std::string& name,
+                const uint8_t* data, const uint64_t size )
+{
+    std::unique_ptr< pression::data::Compressor > compressor( info.create( ));
+
+    compressor->compress( data, size );
+    lunchbox::Clock clock;
+    const auto& compressed = compressor->compress( data, size );
+    const float compressTime = clock.getTimef();
+
+    pression::data::Compressor::Result result( size );
+    compressor->decompress( compressed, result.getData(), size );
+
+    clock.reset();
+    compressor->decompress( compressed, result.getData(), size );
```

Comment:
Both compress/decompress get one untimed 'dry run' to touch the memory once. This delivers more realistic results with the expected (re)use in applications.